2-0-data-structures-and-algorithms/src/test/java/com/bobocode/bst/BinarySearchTreeTest.java

```diff
@@ -1,82 +1,249 @@
 package com.bobocode.bst;
 
-import org.junit.jupiter.api.*;
-import org.junit.jupiter.params.*;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.function.Predicate;
 import java.util.stream.Stream;
 
 import static org.assertj.core.api.Assertions.*;
 import static org.junit.jupiter.params.provider.Arguments.arguments;
 
-
+@Slf4j
 @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 class BinarySearchTreeTest {
 
+    private static final Predicate<Field> SIZE_FIELD = field -> field.getName().equals("size") || field.getName().equals("length");
+    private static final Predicate<Field> NODE_FIELD = field -> field.getType().getSimpleName().equals("Node");
+    private static final Predicate<Field> ELEMENT_FIELD = field -> field.getName().equals("element")
+            || field.getName().equals("item")
+            || field.getName().equals("value");
+    private static final Predicate<Field> LEFT_FIELD = field -> field.getName().equals("left");
```

Comment:
@erneman1 it does not have to be `left` exactly. What if I call my field `leftChild`? You should better use `toLowerCase().contains("left")`